include/service/jobs/sites/site_create_edit_job.js

```diff
@@ -96,7 +93,24 @@ module.exports = function SiteCreateEditJobModule(pb) {
                         site = pb.DocumentCreator.create('site', mySite);
                     }
 
-                    site.hostname = mySite.hostname || site.hostname;
+                    var prevHostname = site.hostname;
+                    var newHostname = mySite.hostname;
+                    if (!site.prevHostnames) {
+                        site.prevHostnames = [];
+                    }
+                    // If this site's hostname has been changed, save off a redirectHost
+                    if ((prevHostname && newHostname) && (prevHostname !== newHostname)) {
+                        // Check for circular hostname references
+                        for (var i=0; i< site.prevHostnames.length; i++) {
+                            if (site.prevHostnames[i] == newHostname) {
+                                site.prevHostnames.splice(i, 1);
+                            }
+                        }
+                        site.prevHostnames.push(prevHostname);
+                        pb.RequestHandler.redirectHosts[prevHostname] = newHostname;
+                        pb.RequestHandler.sites[prevHostname] = null;
+                    }
+                    site.hostname = newHostname || prevHostname;
```

Comment:
This logic would be better suited to live in the ```SiteService.merge``` function.  It would isolate the assignment of fields to the service that owns the object.

Depending on how out of scope you'd like to get... You don't actually need to call to load the site via the call to ```dao.loadById```.  The save method will do this automatically if the ```_id``` property is set on "mySite".  All you have to do is pass the raw ```mySite``` object to save and let the SiteService handle the rest.